packages/x-charts/src/BarChart/BarLabel/BarLabel.tsx

```diff
@@ -57,6 +57,12 @@ export type BarLabelProps = Omit<
      * Height of the bar this label belongs to.
      */
     height: number;
+    /**
+     * The placement of the bar label.
+     * It controls whether the label is placed inside the bar or above it.
+     * @default 'inside'
+     */
+    barLabelPlacement?: 'inside' | 'above';
```

Comment:
For now these options are enough, but I'm considering if we should prepare for more options.

I'm thinking that users might want to position a bar label above the top of the bar, but also below it. 

I was thinking of an API like `barLabelPlacement: 'inside-top' | 'outside-top' | '...'`. @JCQuintas @alexfauquette any opinions?

Reply:
`barLabelPlacement: 'inside-top' | 'outside-top' | 'inside-bottom' | 'outside-bottom' ` for each series should possibly cover all scenarios including positive, negative, horizontal, vertical bars.